### ISSUE TITLE
`#[derive(Decode)]` fails when struct fields are raw identifiers

### ISSUE THREAD

**Issue by boringcactus** [tailhook/knuffel#2]
```rust
#[derive(knuffel::Decode)]
struct TestCase {
    #[knuffel(argument)]
    r#ref: String,
}
```
causes the `Decode` macro to panic with `` `"seen_r#ref"` is not a valid identifier``.

**Comment by tailhook** [comment]
Oh, this is interesting. Should be easy to fix. I.e. whenever we create a slightly different than original identifier we should somehow take care of raw idents (there should be some helper function in `syn` I think).

Do you want to contribute?

**Comment by boringcactus** [comment]
I took a look at it but couldn't quite get anywhere - proc macro debugging is a bit of a mess - but the helper function you're looking for is [`IdentExt::unraw`](https://docs.rs/syn/1.0.89/syn/ext/trait.IdentExt.html#tymethod.unraw).

**Comment by tailhook** [comment]
I think it should be possible to use `quote::format_ident!` instead of `syn::Ident::new(&format!("seen_{}", fld),` (and few similar places), if you want to try.

**Comment by tailhook** [comment]
Thanks! Released as 2.0 (along with miette upgrade)
